Source/Models/SwishError.swift

```diff
@@ -2,47 +2,69 @@ import Foundation
 import Argo
 
 public enum SwishError {
-  case ArgoError(Argo.DecodeError)
-  case DeserializationError(NSError)
-  case ParseError(NSError)
-  case ServerError(code: Int, data: NSData?)
-  case URLSessionError(NSError)
+  case argoError(Argo.DecodeError)
+  case deserializationError(Error)
+  case parseError(Error)
+  case serverError(code: Int, data: Data?)
+  case urlSessionError(Error)
 }
 
-extension SwishError: ErrorType { }
+extension SwishError: Error { }
 
-public extension SwishError {
-  var rawError: NSError {
+extension SwishError: CustomNSError {
+  public static let errorDomain = "com.thoughtbot.swish"
+
+  public var errorCode: Int {
+    switch self {
+    case .argoError:              return 1
+    case .deserializationError:   return 2
+    case .parseError:             return 3
+    case .serverError:            return 4
+    case .urlSessionError:        return 5
+    }
+  }
+
+#if swift(>=3.1)
+#else
+  public var errorUserInfo: [String: Any] {
+    var userInfo: [String: Any] = [:]
+    if let errorDescription = errorDescription {
+      userInfo[NSLocalizedDescriptionKey] = errorDescription
+    }
```

Comment:
I think you can remove this unwrapping and set it explicitly because [`[]` accepts an optional](https://developer.apple.com/reference/swift/dictionary/1540848-subscript)?